### ISSUE TITLE
Remove Eloquent\Model::jsonSerialize() return type

### ISSUE THREAD

**Issue by mikewudev** [laravel/framework#42121]
- Laravel Version: 9.9.0
- PHP Version: 8.1.5
- Database Driver & Version: Mysql 5.7

### Description:

TL;DR:

`Model::jsonSerialize()` Now has a return type of `array` which will throw a PHP 8 `TypeError` when returning a `string`. Can we remove this return type, and allow `jsonSerialize` to return strings? 

Background:

I've been overriding `toArray()` to return a string for simple models that only contain a single attribute (ie., `name`) to let Laravel automatically serialize it to a string. Works great for eloquent relationships. Unfortunately this is now broken in Laravel 9 (PHP 8), and I'd like to keep the existing behavior (or find a better alternative without having to serialize it manually?)

 
### Steps To Reproduce:

Override `toArray()` in any model, and return a `string`.

```php
   public function toArray()
    {
        return $this->name;
    }
```

**Comment by driesvints** [comment]
@mikewudev `toArray` can never return a string. That's a bug in your code.

**Comment by mikewudev** [comment]
Thanks for the reply @driesvints . Yea, I've been incorrectly overriding `toArray`, where I should have been replacing `toJson`
